### PR TITLE
#289: Flag redundant `let/var` in optional binding

### LINKED ISSUE #289: Flag consecutive `let/var` usages in optional binding
**Preferred**
```swift
if var a = a, b = b, c = c where c != 0 {
    print("(a + b) / c = \((a + b) / c)")     // (a + b) / c = 5
}
```

**Not Preferred**
```swift
if var a = a, var b = b, c = c where c != 0 {
    print("(a + b) / c = \((a + b) / c)")     // (a + b) / c = 5
}

if let a = a, b = b, let c = c where c != 0 {
    print("(a + b) / c = \((a + b) / c)")     // (a + b) / c = 5
}
```

### PR DESCRIPTION
**Pull request by adityatrivedi**
Resolves #289.

### CODE REVIEW

File: src/test/java/com/sleekbyte/tailor/functional/RedundantOptionalBindingTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    private void addExpectedMessage(int line, int column, String construct) {
+        this.expectedMessages.add(
+            Printer.genOutputStringForTest(
+                Rules.REDUNDANT_OPTIONAL_BINDING,
+                inputFile.getName(),
+                line,
+                column,
+                Severity.WARNING,
+                Messages.CONSECUTIVE + "'" + construct + "' " + Messages.AT_COLUMN + column
```
Comment (a4sriniv) [inline src/test/java/com/sleekbyte/tailor/functional/RedundantOptionalBindingTest.java]: Hmm, can we just say `'var' at column 26 is redundant and should be removed`? I know it goes against our usual style, but it makes more sense than using 'consecutive' which can be misleading and doesn't really point out the actual problem.

Comment (alykhank) [inline src/test/java/com/sleekbyte/tailor/functional/RedundantOptionalBindingTest.java]: Shouldn't we be bringing attention to the fact that the `var`/`let` is repeated immediately after it was used in the previous optional binding? That is the reason for the redundancy.

**Comment by alykhank** [comment]
:+1:

**Comment by a4sriniv** [comment]
:+1:

**Pull request merged**